migration/incremental/032-remove-non-empty-span-name-constraint.sql

```diff
@@ -0,0 +1,14 @@
+-- Our tracing implementation is based on OTLP spec, though OTLP spec
+-- doesn't mandate span name to be non empty[1], we have a constraint
+-- in the _ps_trace.operation table which raises error on empty span name.
+-- On the other hand, Jaeger span name can be empty and their storage
+-- integration tests validates the same[2]. Jaeger author even suggested to
+-- remove[3] the constraint as it shouldn't trouble any part of the system.
+-- [1] https://github.com/open-telemetry/opentelemetry-proto/blob/724e427879e3d2bae2edc0218fff06e37b9eb46e/opentelemetry/proto/trace/v1/trace.proto#L110-L121
+-- [2] https://github.com/jaegertracing/jaeger/blob/7872d1b07439c3f2d316065b1fd53e885b26a66f/plugin/storage/integration/fixtures/traces/default.json#L6
+-- [3] https://github.com/jaegertracing/jaeger/pull/3922#issuecomment-1256505785
+
+-- `_ps_trace.operation` table has been created with `span_name text NOT NULL CHECK (span_name != '')`, we just want to remove the `CHECK(span_name != '')`.
+-- Ideally we should have used 'ALTER TABLE <table> ALTER COLUMN <column> DROP NOT NULL', but it is not working with a column having 'CHECK' constraint, hence dropping the CONSTRAINT using implicit name and readding `NOT NULL` CONSTRAINT.
+ALTER TABLE _ps_trace.operation DROP CONSTRAINT operation_span_name_check;
+ALTER TABLE _ps_trace.operation ALTER COLUMN span_name SET NOT NULL;
```

Comment:
why would the set not null be necessary. Isn't the column already NOT NULL?
```suggestion

```

Reply:
![image](https://user-images.githubusercontent.com/4974668/192796724-f20643f0-9bfc-4d0b-9c08-cfd1c018d3cd.png)

It appears to me that dropping the check constraint does not impact the not null constraint. I think we should remove line 14.